crates/wasmtime/src/module.rs

```diff
@@ -80,8 +80,10 @@ pub struct Module {
 
 struct ModuleInner {
```

Comment:
The `Store` here internally looks to be the only reason that this is non-`Send`, right? If that's the case can we move the `Store` out, or better yet only take an `Engine` to compile a `module`?

This would require taking a `Store` argument to create an `Instance` but I think that's ok since `Linker` is the "ergonomic" way to create instances.

Additionally this would require, on instantiation, registration of JIT code ranges within a `Store` if a module hasn't previously been registered there, but I think that's ok?

Reply:
I redesigned API to remove `Store` from `Module`. @alexcrichton can you take a look at this change before I proceed any further?